app.py

```diff
@@ -102,6 +102,10 @@ def render_and_submit_version(self, template, fields, first_frame, last_frame, s
         output_path_template = self.get_template("movie_path_template")
         output_path = output_path_template.apply_fields(fields)
 
+        # Get the name for the Version entity
+        version_template = self.get_template("sg_version_name_template")
+        version_name = version_template.apply_fields(fields)
```

Comment:
I suspect that if the `sg_version_name_template` setting is empty that the `self.get_template` call above will return a `None`, which will then fail ungracefully here when apply_fields is called. Could you try that situation out to confirm, and if I'm correct set this up so that version_name ends up as a `None`? I believe that will work fine with the changes you've made to submit_version later in the submitter.